### PR TITLE
align window.setStatusBarMessage with vscode

### LINKED ISSUE #4759: Multiple status bar entries from rust vscode extension if tooling is not installed.
Screen Shot 2019-03-29 at 12 01 47

Extension: https://github.com/rust-lang/rls-vscode

### PR DESCRIPTION
**Pull request by akosyakov**
fix #4759, basically copies impl from VS Code

In order to verify:
- install rust extension
- check that only status message from RLS appears, not like in #4759

I'm using this repo https://github.com/geropl/ps-rs.git

TODO:
- [x] open a CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19523

### CODE REVIEW

File: packages/plugin-ext/src/plugin/status-bar-message-registry.ts [hunk 1]
```diff
@@ -21,39 +21,82 @@ import {
...
     proxy: StatusBarMessageRegistryMain;
 
+    protected readonly statusMessage: StatusBarMessage;
+
     constructor(rpc: RPCProtocol) {
         this.proxy = rpc.getProxy(Ext.STATUS_BAR_MESSAGE_REGISTRY_MAIN);
+        this.statusMessage = new StatusBarMessage(this);
     }
 
+    // copied from https://github.com/Microsoft/vscode/blob/6c8f02b41db9ae5c4d15df767d47755e5c73b9d5/src/vs/workbench/api/node/extHostStatusBar.ts#L122
```
Comment (olexii4) [inline packages/plugin-ext/src/plugin/status-bar-message-registry.ts]: Is it typo?
L122 -> L174

Comment (akosyakov) [reply packages/plugin-ext/src/plugin/status-bar-message-registry.ts]: Is not obvious what was copied? I can add L174 if it is not.

Comment (akosyakov) [reply packages/plugin-ext/src/plugin/status-bar-message-registry.ts]: Ah, understood!

Comment (akosyakov) [reply packages/plugin-ext/src/plugin/status-bar-message-registry.ts]: fixed!

**Pull request merged**